pkg/security/security_profile/activity_tree/activity_tree.go

```diff
@@ -970,6 +970,16 @@ func (at *ActivityTree) EvictUnusedNodes(before time.Time) int {
 		if node == nil {
 			continue
 		}
+
+		// Try a fallback if the node is in the process cache
+		if filepathsInProcessCache[node.Process.FileEvent.PathnameStr] {
+			// check if the node was supposed to be removed, then update the last seen to now
+			evictableImageTags := node.NodeBase.GetEvictableImageTags(before)
+			for _, imageTag := range evictableImageTags {
+				node.NodeBase.AppendImageTag(imageTag, time.Now())
+			}
```

Comment:
IMHO we should do this BY image tag, otherwise we'll keep refresh process nodes for version that we didn't see for a long time. The approach is good but instead of a map[string]bool we may want to pass a map[string]map[string]bool to have the filepaths by image tags.
Even that is not perfect: we won't care about lineage. I wonder if it's an issue or not. Like if we have a process tree like `foo->bar->foo`, if the second `foo` has not been seen since a long time, it still will be refreshed by this new logic. If we can find a better option it will be great, without over-engeenering it :)